lib/cisco_node_utils/pim.rb

```diff
@@ -0,0 +1,102 @@
+# NXAPI implementation of the Pim class
+# Provides configuration of PIM and its various properties
+# like ssm-range, etc
+#
+# Smitha Gopalan, November 2015
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#-------------------------------------------------------
+# CLI: <afi> pim ssm-range <range>  (under different VRFs)
+#-------------------------------------------------------
+
+require_relative 'node_util'
+
+module Cisco
+  # node_utils class for Pim
+  class Pim < NodeUtil
+    attr_reader :vrf, :afi
+
+    # Constructor with vrf
+    # ---------------------
+    def initialize(afi, vrf='default', instantiate=true)
+      fail ArgumentError unless vrf.is_a? String
+      fail ArgumentError unless vrf.length > 0
+      @vrf = vrf
+      @afi = Pim.afi_cli(afi)
+      if @vrf == 'default'
+        @get_args = @set_args = { state: '', afi: @afi }
+      else
+        @get_args = @set_args = { state: '', vrf: @vrf, afi: @afi }
+      end
```

Comment:
Don't set `state: ''` as part of the initial class var assignment. The setters take care of that. 

Also, your `set_args_keys_default` already takes care of the vrf logic so you don't need to do this here. Just do this instead:

```
-      if @vrf == 'default'
-        @get_args = @set_args = { state: '', afi: @afi }
-      else
-        @get_args = @set_args = { state: '', vrf: @vrf, afi: @afi }
-      end
+    set_args_keys_default
```